Marlin/src/feature/easythreed_ui.cpp

```diff
@@ -188,46 +196,50 @@ void EasythreedUI::printButton() {
             }
-            card.ls();                                            // List all files to serial output
-            const uint16_t filecnt = card.countFilesInWorkDir();  // Count printable files in cwd
-            if (filecnt == 0) return;                             // None are printable?
-            card.selectFileByIndex(filecnt);                      // Select the last file according to current sort options
-            card.openAndPrintFile(card.filename);                 // Start printing it
             break;
           }
-          case PF_PAUSE: {                                          // Pause printing (not currently firing)
-            if (!printingIsActive()) break;
-            blink_interval_ms = LED_ON;                             // Set indicator to steady ON
-            queue.inject(F("M25"));                                 // Queue Pause
-            print_key_flag = PF_RESUME;                             // The "Print" button now resumes the print
-            break;
+          case PF_PAUSE: {                                          // Pause printing
+            if (IS_SD_PRINTING()) {                                 // If pause is during heatup, heatup will continue and then pause before printing
+              card.pauseSDPrint();                                  // This is more reliable than queueing M25 ??
+              blink_interval_ms = LED_BLINK_5;                      // Set indicator to steady ON
+              print_key_flag = PF_RESUME;                           // The "Print" button now resumes the print
             }
+            break;
           case PF_RESUME: {                                         // Resume printing
-            if (printingIsActive()) break;
-            blink_interval_ms = LED_BLINK_2;                        // Blink the indicator LED at 1 second intervals
-            queue.inject(F("M24"));                                 // Queue resume
-            print_key_flag = PF_PAUSE;                              // The "Print" button now pauses the print
+            if (IS_SD_PAUSED()) {
+              card.startOrResumeFilePrinting();                     // This is more reliable than queueing M24 ??
+              blink_interval_ms = LED_BLINK_2;                      // Blink the indicator LED at 1 second intervals
+              print_key_flag = PF_PAUSE;                            // The "Print" button now pauses the print
+            }
             break;
           }
         }
       }
       else {                                                        // Register a longer press
         if (print_key_flag == PF_START && !printingIsActive())  {   // While not printing, this moves Z up 10mm
           blink_interval_ms = LED_ON;
-          queue.inject(F("G91\nG0 Z10 F600\nG90"));                 // Raise Z soon after returning to main loop
+          queue.inject(F("G91\nG0Z10F600\nG90\nM18"));              // Raise Z soon after returning to main loop
```

Comment:
On some machines the bed will fall when the power is disabled. There are options to mitigate this for those kinds of setups.